index.html

```diff
@@ -2888,11 +2900,59 @@ <h3><dfn>Writing or pushing content</dfn></h3>
       </p>
       </section>
 
+      <section><h3>Mapping external data to NDEF</h3>
+      <p>
+        To <dfn>map external data to NDEF</dfn> given a |record:NDEFRecordInit|,
+        run these steps:
+        <ol class=algorithm>
+          <li>
+            If the type of a |record|'s data is not an
+            {{ArrayBuffer}}, [= exception/throw =] a {{TypeError}}
+            and abort these steps.
+          </li>
+          <div class=issue>
+            In addition to supporting ArrayBuffer, we need to support sub records. That
+            is really what makes external records interesting.
+          </div>
+          <li>
+            Set |arrayBuffer| to |record|'s data.
+          </li>
+          <li>
+            Set |length:unsigned long| to |arrayBuffer|.[[\ArrayBufferByteLength]].
+          </li>
+          <li>
+            Set |data:byte sequence| to |arrayBuffer|.[[\ArrayBufferData]].
+          </li>
+          <li>
+            Let |ndefRecord| be the notation for the <a>NDEF record</a> to
+            be created by the UA.
+            <ol>
+              <li>
+                Set |ndefRecord|'s <a>TNF field</a> to `4` (<a>external type record</a>).
+              </li>
+              <li>
+                Set the |ndefRecord|'s <a>TYPE field</a> to |record|'s recordType.
```

Comment:
I guess this is tricky, too. No checks needed?

Reply:
if it gets here it should follow the ABNF, so all should be correct